File: patterns/javascript-patterns/pattern105.js
// Pattern 105: ...

// *     * * * *
// *     *
// *     *
// * * * * * * *
//       *     *
//       *     *
// * * * *     *

let n = 7;

for (let i = 1; i <= n; i++) {
    let row = "";
    for (let j = 1; j <= n; j++) {
        if (
            i === Math.floor(n / 2) + 1 ||
            j === Math.floor(n / 2) + 1 ||
            (i === 1 && j >= Math.floor(n / 2) + 1) ||
            (i === n && j <= Math.floor(n / 2)) ||
            (j === 1 && i <= Math.floor(n / 2)) ||
            (j === n && i >= Math.floor(n / 2) + 1)
        ) {
            row += "* ";
        } else {
            row += "  ";
        }
    }
    console.log(row);
}

File: patterns/python-patterns/pattern105.py
# Pattern 105: ...

# *     * * * *
# *     *
# *     *
# * * * * * * *
#       *     *
#       *     *
# * * * *     *

n = 7

for i in range(1, n + 1):
    for j in range(1, n + 1):
        if i == n // 2 + 1 or j == n // 2 + 1 or \
           (i == 1 and j >= n // 2 + 1) or \
           (i == n and j <= n // 2) or \
           (j == 1 and i <= n // 2) or \
           (j == n and i >= n // 2 + 1):
            print("* ", end="")
        else:
            print("  ", end="")
    print()
